chalice/deploy/deployer.py

```diff
@@ -421,11 +421,25 @@ def deploy(self, config, existing_resources, stage_name):
                                    deployed_values)
         self._deploy_event_sources(config, existing_resources, stage_name,
                                    deployed_values)
+        self._deploy_pure_lambda_functions(config, existing_resources,
+                                           stage_name, deployed_values)
         if existing_resources is not None:
             self._cleanup_unreferenced_functions(existing_resources,
                                                  deployed_values)
         return deployed_values
 
+    def _deploy_pure_lambda_functions(self, config, existing_resources,
```

Comment:
From looking at this logic it seems I can create two pure lambda functions with the same name, and the deployment will crash out with a ResourceAlreadyExistsException. Is that correct? Can we do some validation like we do on routes here we check that they are unique if this is the case?